packages/web3-shared/src/utils/formatter.ts

```diff
@@ -14,8 +14,8 @@ export function formatPrice(price: BigNumber.Value, decimalPlaces = 6) {
     return new BigNumber(price).decimalPlaces(decimalPlaces).toString()
 }
 
-export function formatAmount(amount: BigNumber.Value = '0', decimals = 0) {
-    return new BigNumber(amount).multipliedBy(pow10(decimals)).toFixed()
+export function formatAmount(amount: BigNumber.Value, decimals = 0) {
+    return new BigNumber(amount || '0').multipliedBy(pow10(decimals)).toFixed()
```

Comment:
why?

Reply:
If `amount` from user input is empty string `''`, here will result `NaN`.